alembic/script/revision.py

```diff
@@ -156,12 +189,49 @@ def _revision_map(self):
                 down_revision = map_[downrev]
                 down_revision.add_nextrev(rev)
                 if downrev in rev._versioned_down_revisions:
-                    heads.discard(downrev)
-                _real_heads.discard(downrev)
+                    heads.discard(down_revision)
+                _real_heads.discard(down_revision)
+
```

Comment:
**Koichiro Den** (lkpdn) wrote:

Done. PTAL at Patch Set 3.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/2379